tdms/CMakeLists.txt

```
@@ -136,20 +136,9 @@ install(TARGETS tdms)
 if (BUILD_TESTING)
     enable_testing()
 
-    # find all the array.h dataype tests
-    file(GLOB ARRAY_TEST_FILES
-        "${CMAKE_SOURCE_DIR}/tests/unit/array_tests/test_*.cpp"
-    )
-
-    add_executable(tdms_tests
-            tests/unit/test_fields.cpp
-            tests/unit/test_BLi_vs_cubic_interpolation.cpp
-            tests/unit/test_field_interpolation.cpp
-            tests/unit/test_interpolation_determination.cpp
-            tests/unit/test_interpolation_functions.cpp
-            tests/unit/test_numerical_derivative.cpp
-            tests/unit/test_openandorder.cpp
-            ${ARRAY_TEST_FILES})
+    # find all .cpp files in the unit tests directory (and subdirectories).
+    file(GLOB_RECURSE UNIT_TEST_SRC "${CMAKE_SOURCE_DIR}/tests/unit/*.cpp")
```

```suggestion
    file(GLOB_RECURSE UNIT_TEST_SRC "${CMAKE_SOURCE_DIR}/tests/unit/test_*.cpp")
```

I doubt we'll ever populate these folders with anything other than tests, but why not enforce a tradition we've started?

---

Oooh. Nice. I wasn't expecting that to work, but it does.